helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/WagedRebalancer.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.helix.controller.rebalancer.util;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Collection;
+
+import org.apache.helix.model.ResourceAssignment;
+
+
+public class WagedRebalancer {
```

Comment:
Why is this a separate method in a util class?

1) why is this named WagedRebalancer? This is not WagedRebalancer. 2) I don't think there's enough justification for making a util class for one static method. Could you perhaps embed it in the metric-calculation logic?

Reply:
Put it in a new util class to separate and reduce the code in WagedRebalancer. It is not WagedRebalancer. This is a util class related to WagedRebalancer. Maybe I could put it in ResourceUsage to narrow down the scope.

Reply:
Either way, I get the feeling that creating a util class just for this is not necessary. Also just to share another thought, if you really needed a util class, you probably want to name it WagedRebalancerUtil, or better, WagedRebalancerMetricCalcuationUtil. The point I'm trying to get across here is that it shouldn't be called WagedRebalancer.

Could you please try to contain this logic in some metric-related class?